HoneyDo.Web/ClientApp/src/components/Tasks/Form.tsx

```diff
@@ -128,6 +129,13 @@ class TaskForm extends Component<TaskFormProps, TaskFormState> {
           <Button className={classes.button} onClick={() => onCancel()}>
             Cancel
           </Button>
+          {task ? (
+            <Button
+              className={classes.button}
+              onClick={() => (onDelete ? onDelete() : null)}>
```

Comment:
I haven't run this yet (just on my desktop). Why is there a prop called `onDelete` being passed and if it's not passed, we can't delete?
Also it seems this is a good place for an enhancement as it will return true or false. The form should have a callback that updates the parent of the deletion, update, or created status and the appropriate modal, warning, or toast should display.

Reply:
This is bubbling up an optional event. Optional because if you are creating a new task you don’t want the option to delete.  In that case the parent component doesn’t pass the “onDelete” prop and the form won’t render the button.

Reply:
On second thought, this should be changed to require the “task” prop and the “onDelete” to render the delete button.

Reply:
Since you are creating a new task, the `task` doesn't exist yet so it doesn't render the delete button. That works!